gsy_e_sdk/setups/Asset_API_scripts/degrees_of_freedom.py

```diff
@@ -0,0 +1,138 @@
+"""
+Template file for the implementation of the Degrees of
+Freedom through the gsy-e-sdk api client using Redis.
+"""
+
+import os
+import json
+from time import sleep
+from typing import List, Dict
+from gsy_e_sdk.redis_aggregator import RedisAggregator
+from gsy_e_sdk.clients.redis_asset_client import RedisAssetClient
+
+module_dir = os.path.dirname(__file__)
+ORACLE_NAME = "oracle"
+
+# List of assets's names to be connected with the API
+LOAD_NAMES = ["Load 1 L13", "Load 2 L21", "Load 3 L17"]
+PV_NAMES = ["PV 9 (15kW)", "PV 5 (10kW)"]
+
+
+class Oracle(RedisAggregator):
+    """Class that defines the behaviour of an "oracle" aggregator."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.is_finished = False
+        self.degrees_of_freedom = {}
+
+    def on_market_cycle(self, market_info):
+        """Place a bid or an offer whenever a new market is created."""
+        if self.is_finished is True:
+            return
+        self.read_requirements()
```

Comment:
Nit: Right now, we're reading the requirements file on every market cycle event. I think this could be moved in the `__init__()` method so that we only read the DoF requirements once when the Oracle is instantiated.